pkg/graveler/committed/apply.go

```diff
@@ -157,7 +163,7 @@ func (a *applier) apply() error {
 		}
 	}
 
-	if !a.opts.AllowEmpty && !a.hasChanges(a.summary) {
+	if !a.opts.AllowEmpty && !a.hasChanges(a.summary) && !a.summary.MissingInfo {
```

Comment:
There is an assumption here that - if there is missing info, then the optimization was used, which means there _are_ changes, right?
Can you add a comment to explain that?

Reply:
Moved to `hasChanges` and documented. Thanks